File: kt/main/kotlin/moduleconfig/pushtovar/fs/fooSfl.kt
/*
 *  Copyright © 2021 Paulo Villela. All rights reserved.
 *  Use of this source code is governed by the MIT license
 *  that can be found in the LICENSE file.
 */

package tryout.moduleconfig.pushtovar.fs

import tryout.moduleconfig.pushtovar.fwk.CfgSrc

data class FooSflCfgInfo(
	val x: String
)

val fooSflCfgSrc = CfgSrc<FooSflCfgInfo>()

fun fooSfl() {
	println(fooSflCfgSrc().x)
	barBf()
}

File: ts/src/push-to-var/fs/foo-sfl.ts
/*
 * Copyright © 2022 Paulo Villela. All rights reserved.
 * Use of this source code is governed by the MIT license
 * that can be found in the LICENSE file.
 */

import { makeCfgSrc } from "../fwk/cfg-src";
import { barBf } from "./bar-bf";

export type FooSflCfgInfo = {
  x: string;
};

export const fooSflCfgSrc = makeCfgSrc<FooSflCfgInfo>(null);

export function fooSfl() {
  console.log(fooSflCfgSrc().x);
  barBf()
}
